Rear Rider/Rear Rider/Auth/SignInView.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  SignInView.swift
+//  timerr
+//
+//  Created by Robert Paul Sutton on 1/27/22.
+//
+
+import SwiftUI
+import Firebase
+
+struct SignInView: View {
+    @EnvironmentObject var auth: AuthModel
+    @EnvironmentObject var viewRouter: ViewRouter
+    @State var email = ""
+    @State var password = ""
+    @State var signInErrorMessage = ""
+    
+    var body: some View {
+        VStack(spacing: 15) {
+            Text("Rear Rider").font(.title)
+            SignInCredentialFields(email: $email, password: $password)
+            HStack {
+                Spacer()
+                Button(action: {
+                    viewRouter.currentPage = .forgotPasswordPage
+                }) {
+                    Spacer()
+                    Text("Forgot Password?")
+                }
+            }
+            Button(action: {
+                signIn()
+            }) {
+                Text("Log In")
+                    .bold()
+                    .frame(width: 360, height: 50)
+                    .background(.blue)
+                    .foregroundColor(.white)
+                    .cornerRadius(10)
+                    .opacity(!auth.authLoading && !email.isEmpty && !password.isEmpty ? 0.9 : 0.4)
+            }
+            .disabled(!auth.authLoading && !email.isEmpty && !password.isEmpty ? false : true)
```

Comment:
Checking on an OR and flipping your condition here would be a little more readable and slightly more performant:

```suggestion
                    .opacity(auth.authLoading || email.isEmpty || password.isEmpty ? 0.4 : 0.9)
            }
            .disabled(auth.authLoading || email.isEmpty || password.isEmpty ? true : false)
```